src/app/rundown-view/factories/key-binding.factory.ts

```diff
@@ -198,6 +204,24 @@ export class KeyBindingFactory {
     }
   }
 
+  private cycleMiniShelfForward(rundown: Rundown): void {
+    try {
+      this.miniShelfCycleService.cycleMiniShelfForward(rundown)
+    } catch (error) {
+      this.logger.data(error).error('Failed cycling forward in mini shelves.')
+      this.snackBar.open('Unable to cycle forward in mini shelves.', undefined, { panelClass: 'snackbar-danger', duration: 3000 })
+    }
+  }
+
+  private cycleMiniShelfBackward(rundown: Rundown): void {
+    try {
+      this.miniShelfCycleService.cycleMiniShelfBackward(rundown)
+    } catch (error) {
+      this.logger.data(error).error('Failed cycling backward in mini shelves.')
+      this.snackBar.open('Unable to cycle backward in mini shelves.', undefined, { panelClass: 'snackbar-danger', duration: 3000 })
+    }
+  }
```

Comment:
The error handling for cycling the miniShelf does not belong in the `KeyBindingFactory`. This factory should just bind a key to something and that something should handle the rest.

The responsibility of error handling for cycling MiniShelf belongs in the `MiniSehlfCycleService`.

This means that all the KeyBindingFactory should do is:
```
this.createRundownKeyBinding('Cycle MiniShelf', ['Tab'], () => this.miniShelfCycleService.cycleMiniShelfForward(rundown)),
this.createRundownKeyBinding('Cycle MiniShelf', ['Shift', 'Tab'], () => this.miniShelfCycleService.cycleMiniShelfBackward(rundown)),
```

By moving the error handling to `CycleMiniShelfService` also means that we don't need to re-implemt the error handling if we want to trigger the cycling from somewhere else in the system.